### PR TITLE
Fixes #9858 - monkey patch for older versions of Foreman

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by ares** [comment]
replaces #2265

File: lib/ssl_context_monkey_patch.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require 'openssl'
+
+# we need to enforce verify_mode => 1 on SSL context
+# more info in http://projects.theforeman.org/issues/9858
+class OpenSSL::SSL::SSLContext
+  alias __original_initialize initialize
+  private :__original_initialize
+
+  def initialize(*args)
+      __original_initialize(*args)
```
Comment (domcleal) [inline lib/ssl_context_monkey_patch.rb]: indentation's off here

File: lib/ssl_context_monkey_patch.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require 'openssl'
+
+# we need to enforce verify_mode => 1 on SSL context
+# more info in http://projects.theforeman.org/issues/9858
+class OpenSSL::SSL::SSLContext
+  alias __original_initialize initialize
```
Comment (domcleal) [inline lib/ssl_context_monkey_patch.rb]: alias_method_chain?

File: config/application.rb [hunk 3]
```diff
@@ -42,6 +42,7 @@
 require File.expand_path('../../lib/timed_cached_store.rb', __FILE__)
 require File.expand_path('../../lib/foreman/exception', __FILE__)
 require File.expand_path('../../lib/core_extensions', __FILE__)
+require File.expand_path('../../lib/ssl_context_monkey_patch', __FILE__)
```
Comment (domcleal) [inline config/application.rb:45]: perhaps better to put in core_extensions?

**Comment by domcleal** [comment]
Oh, apparently it doesn't test.  Apologies.  Couple of nitpicky things above.

File: config/application.rb (hunk 3 above)
Comment (ares) [inline config/application.rb:45]: I prefer to make every monkey patch as visible as possible

**Comment by ares** [comment]
comments fixed except one, could you please give it another try?

**Comment by domcleal** [comment]
Thanks @ares, merged as 74f2d0e for Foreman 1.7.4.

**Pull request closed without merging**